### PR TITLE
Feature/map location

### LINKED ISSUE #41: Implement own location in the map
#38

### PR DESCRIPTION
**Pull request by sirpudika**
Adds location service to the map (incl. permission demand). Shows location which is the Google headquarter by default.

### CODE REVIEW

File: app/src/androidTest/java/com/github/gogetters/letsgo/utils/PermissionUtilsTest.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+class PermissionUtilsTest {
+
+    @Test
+    fun requestPermissionTest() {
+        // TODO implement
+        assertTrue(true)
+    }
+
+    @Test
+    fun isPermissionGrantedWorksWithoutPermission() {
```
Comment (lupusmalus) [inline app/src/androidTest/java/com/github/gogetters/letsgo/utils/PermissionUtilsTest.kt]: I would change the name to make it more evident that the test should reject (e.g.) rejectIfNoPermission

File: app/src/androidTest/java/com/github/gogetters/letsgo/utils/PermissionUtilsTest.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+        // non-empty case
+        var permissions: Array<String> = arrayOf(Manifest.permission.ACCESS_FINE_LOCATION)
+        var permissionResults: IntArray = IntArray(2)
+        permissionResults.set(0, PackageManager.PERMISSION_GRANTED)
+        assertFalse(isPermissionGranted(permissions, permissionResults, Manifest.permission.ACCESS_COARSE_LOCATION))
+    }
+
+    @Test
+    fun isPermissionGrantedWorksWithPermission() {
```
Comment (lupusmalus) [inline app/src/androidTest/java/com/github/gogetters/letsgo/utils/PermissionUtilsTest.kt]: Consequently might want to change the name as well fo fit with line 22

File: app/src/main/java/com/github/gogetters/letsgo/MapsActivity.kt [hunk 3]
```diff
@@ -10,9 +21,15 @@ import com.google.android.gms.maps.SupportMapFragment
 import com.google.android.gms.maps.model.LatLng
 import com.google.android.gms.maps.model.MarkerOptions
 
-class MapsActivity : AppCompatActivity(), OnMapReadyCallback {
```
Comment (lupusmalus) [inline app/src/main/java/com/github/gogetters/letsgo/MapsActivity.kt:13]: Write UI tests using espresso in order to achieve more code coverage and ensure correctness

**Review by lupusmalus — COMMENTED** [review]
Overall good, only tests for the mapsactivity missing, try using espresso with certain use cases (sending locations etc.)

**Review by kprotopapas — APPROVED** [review]
Apart from TODOs in production (i do that too...) I like it!

**Review by lupusmalus — APPROVED** [review]
Tests look good to me

**Pull request merged**